FileDescriptorMeter.c

```diff
@@ -33,11 +35,42 @@ static void FileDescriptorMeter_updateValues(Meter* this) {
    /* only print bar for first value */
    this->curItems = 1;
 
-   /* Use maximum value for scaling of bar mode */
-   this->total = this->values[1];
+   /* Use maximum value for scaling of bar mode
+    *
+    * As the plain total value can be very large compared to
+    * the actually used value, this is capped in the following way:
+    *
+    * 1. If the maximum value is below (or equal to) 1<<16, use it directly
+    * 2. If the maximum value is above, use powers of 2 starting at 1<<16 and
+    *    double it until it's larger than 16 times the used file handles
+    *    (capped at the maximum number of files)
+    * 3. If the maximum is effectively unlimited (AKA > 1<<30),
+    *    Do the same as for 2, but cap at 1<<30.
+    */
+   if (this->values[1] <= 1<<16) {
+      this->total = this->values[1];
+   } else {
+      if (this->total < 16 * this->values[0]) {
+         for (this->total = 1<<16; this->total < 16 * this->values[0]; this->total *= 2) {
+            if (this->total >= 1<<30) {
+               break;
+            }
+         }
+      }
+
+      if (this->total > this->values[1]) {
+         this->total = this->values[1];
+      }
+
+      if (this->total > 1<<30) {
+         this->total = 1<<30;
+      }
```

Comment:
Is there a need to check this case (number of used file descriptors greater than `1<<30`)? Although it's rare in practice, it can happen when someone tries to stress their computer.
```c
      if (this->total < this->values[0]) {
         this->total = this->values[0];
      }
```

Reply:
No, there's no need, as that just provides scaling and for that many files the expected behaviour is to just cap at `1<<30`, thus showing a full meter (see comment in the code).

Besides: Having 1<<30 files open will need on the order of `1<<42` KiB (4TiB) assuming each FD needs just one page of memory. And that's not to mention the other overhead involved.

If you have that many files, there won't be much change in meter mode anyway. I think, we are fine here …